src/m365/pp/commands/tenant/tenant-settings-set.ts

```diff
@@ -0,0 +1,257 @@
+    this.#initValidators();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        walkMeOptOut: !!args.options.walkMeOptOut,
+        disableNPSCommentsReachout: !!args.options.disableNPSCommentsReachout,
+        disableNewsletterSendout: !!args.options.disableNewsletterSendout,
+        disableEnvironmentCreationByNonAdminusers: !!args.options.disableEnvironmentCreationByNonAdminusers,
+        disablePortalsCreationByNonAdminusers: !!args.options.disablePortalsCreationByNonAdminusers,
+        disableSurveyFeedback: !!args.options.disableSurveyFeedback,
+        disableTrialEnvironmentCreationByNonAdminusers: !!args.options.disableTrialEnvironmentCreationByNonAdminusers,
+        disableCapacityAllocationByEnvironmentAdmins: !!args.options.disableCapacityAllocationByEnvironmentAdmins,
+        disableSupportTicketsVisibleByAllUsers: !!args.options.disableSupportTicketsVisibleByAllUsers,
+        disableDocsSearch: !!args.options.disableDocsSearch,
+        disableCommunitySearch: !!args.options.disableCommunitySearch,
+        disableBingVideoSearch: !!args.options.disableBingVideoSearch,
+        shareWithColleaguesUserLimit: typeof args.options.shareWithColleaguesUserLimit !== 'undefined',
+        disableShareWithEveryone: !!args.options.disableShareWithEveryone,
+        enableGuestsToMake: !!args.options.enableGuestsToMake,
+        disableAdminDigest: !!args.options.disableAdminDigest,
+        disableDeveloperEnvironmentCreationByNonAdminUsers: !!args.options.disableDeveloperEnvironmentCreationByNonAdminUsers,
+        disableBillingPolicyCreationByNonAdminUsers: !!args.options.disableBillingPolicyCreationByNonAdminUsers,
+        disableChampionsInvitationReachout: !!args.options.disableChampionsInvitationReachout,
+        disableSkillsMatchInvitationReachout: !!args.options.disableSkillsMatchInvitationReachout
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '--walkMeOptOut [walkMeOptOut]',
+        autocomplete: ['true', 'false']
+      },
+      {
+        option: '--disableNPSCommentsReachout [disableNPSCommentsReachout]',
+        autocomplete: ['true', 'false']
+      },
+      {
+        option: '--disableNewsletterSendout [disableNewsletterSendout]',
+        autocomplete: ['true', 'false']
+      },
+      {
+        option: '--disableEnvironmentCreationByNonAdminusers [disableEnvironmentCreationByNonAdminusers]',
+        autocomplete: ['true', 'false']
+      },
+      {
+        option: '--disablePortalsCreationByNonAdminusers [disablePortalsCreationByNonAdminusers]',
+        autocomplete: ['true', 'false']
+      },
+      {
+        option: '--disableSurveyFeedback [disableSurveyFeedback]',
+        autocomplete: ['true', 'false']
+      },
+      {
+        option: '--disableTrialEnvironmentCreationByNonAdminusers [disableTrialEnvironmentCreationByNonAdminusers]',
```

Comment:
My bad sorry, specced it wrong. U of users should be capitalized.
Don't forget to update this in the docs, telemetry, ...